test_src/sljitTest.c

```diff
@@ -11813,34 +11844,45 @@ static void test92(void)
 #endif /* SLJIT_64BIT_ARCHITECTURE */
 	FAILED(buf[19] != -573621, "test92 case 26 failed\n");
 	FAILED(*(sljit_u8*)(buf + 20) != 240, "test92 case 27 failed\n");
-	FAILED(((sljit_u8*)(buf + 20))[1] != 0x55, "test92 case 28 failed\n");
+	FAILED(((sljit_u8*)(buf + 20))[1] != PADBYTE, "test92 case 28 failed\n");
 	FAILED(buf[21] != 192, "test92 case 29 failed\n");
 	FAILED(buf[22] != -5893, "test92 case 30 failed\n");
 	FAILED(buf[23] != 4059, "test92 case 31 failed\n");
 	FAILED(buf[24] != 6359, "test92 case 32 failed\n");
 	FAILED(buf[25] != (sljit_sw)(buf + 23), "test92 case 33 failed\n");
-	FAILED(((sljit_u8*)(buf + 26))[0] != 0x55, "test92 case 34 failed\n");
+#if (!defined SLJIT_ATOMIC_MIN_WIDTH || SLJIT_ATOMIC_MIN_WIDTH <= 8)
+	FAILED(((sljit_u8*)(buf + 26))[0] != PADBYTE, "test92 case 34 failed\n");
 	FAILED(((sljit_u8*)(buf + 26))[1] != 204, "test92 case 35 failed\n");
-	FAILED(((sljit_u8*)(buf + 26))[2] != 0x55, "test92 case 36 failed\n");
+	FAILED(((sljit_u8*)(buf + 26))[2] != PADBYTE, "test92 case 36 failed\n");
 	FAILED(buf[27] != 105, "test92 case 37 failed\n");
-	FAILED(((sljit_u8*)(buf + 28))[1] != 0x55, "test92 case 38 failed\n");
+	FAILED(((sljit_u8*)(buf + 28))[1] != PADBYTE, "test92 case 38 failed\n");
 	FAILED(((sljit_u8*)(buf + 28))[2] != 240, "test92 case 39 failed\n");
-	FAILED(((sljit_u8*)(buf + 28))[3] != 0x55, "test92 case 40 failed\n");
+	FAILED(((sljit_u8*)(buf + 28))[3] != PADBYTE, "test92 case 40 failed\n");
 	FAILED(buf[29] != 13, "test92 case 41 failed\n");
 	FAILED(buf[30] != 0, "test92 case 42 failed\n");
+#endif /* SLJIT_ATOMIC_MIN_WIDTH <= 8 */
+#if (!defined SLJIT_ATOMIC_MIN_WIDTH || SLJIT_ATOMIC_MIN_WIDTH <= 16)
 	FAILED(((sljit_u16*)(buf + 31))[0] != 0x5555, "test92 case 43 failed\n");
 	FAILED(((sljit_u16*)(buf + 31))[1] != 51403, "test92 case 44 failed\n");
 	FAILED(buf[32] != 14876, "test92 case 45 failed\n");
+#endif /* SLJIT_ATOMIC_MIN_WIDTH <= 16 */
 #if (defined SLJIT_64BIT_ARCHITECTURE && SLJIT_64BIT_ARCHITECTURE)
+#if PADBYTE != 0
 	FAILED(((sljit_u32*)(buf + 33))[0] != 0x55555555, "test92 case 46 failed\n");
+#endif
 	FAILED(((sljit_u32*)(buf + 33))[1] != 0xdeadbeef, "test92 case 47 failed\n");
-	FAILED(buf[34] != 0xffffffff, "test92 case 48 failed\n");
+	FAILED(((sljit_u32*)(buf + 34))[0] != 0xffffffff, "test92 case 48 failed\n");
```

Comment:
this change "hides a bug" because loads and moves in 32bit values are sign extended to the full register.

there was an ugly patch that "solves" the issue (at least partially) but I suspect that since it has no external side effects it might be covered by the sljit constrain of ONLY using registers with the correct type from:

https://github.com/zherczeg/sljit/blob/818706a8027312831bc954fbcb0e0d66c20e0748/doc/overview.txt#L41-L44

Reply:
No. ARM64 and x64 zero extends them.

Reply:
but doing zero extension is a feature which is not supported by the instructions we are using in loongarch (which signs extends by default, just like MIPS did at least when that comment was added).

the important part though, is that if coded correctly that sign extension doesn't need to be visible, and is not in this case as both atomic loads and stores ONLY see the lower 32bit part which is all sljit warranties to be accurate.